daemon/oci_linux.go

```diff
@@ -76,10 +77,28 @@ func setResources(s *specs.Spec, r containertypes.Resources) error {
 		specResources.Devices = s.Linux.Resources.Devices
 	}
 
+	if err := setGPUResources(daemon, s, r.GPUs); err != nil {
+		return err
+	}
+
 	s.Linux.Resources = specResources
 	return nil
 }
 
+func setGPUResources(daemon *Daemon, s *specs.Spec, r []containertypes.GPUSet) error {
```

Comment:
Seems like the experimental check should be higher up in the stack, and greatly simplify this function.

Reply:
Ok, and do you prefer passing a boolean to `setResources` instead of the `daemon` object?

Reply:
No, I would just not call `setGPUResources` if the daemon is not experimental.

Reply:
Sure, but I'm talking about the function one step up: `setResources`, this function needs to be called in all cases.

Reply:
Ah... a little gross but... maybe we can do `setResources` and `setResourcesExperimental`?

Reply:
seems reasonable.